### PR TITLE
Fix deleting inner-map

### PR DESCRIPTION
**Pull request by ksauzz**
This PR fixes that deleting inner item of maps fails.

Test code:

```ruby
#!/usr/bin/env ruby

require 'riak'

client = Riak::Client.new
bucket_type = client.bucket_type 'maps'
bucket = bucket_type.bucket 'test'
key = '1'

map = Riak::Crdt::Map.new bucket, key
map.maps['bag'].maps['123'].registers['name'] = 'f1'
map.maps['bag'].maps.delete('123')
```

Error message:

```
/Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/beefcake-1.1.0/lib/beefcake.rb:290:in `block in assign': undefined method `map' for "123":String (NoMethodError)
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/beefcake-1.1.0/lib/beefcake.rb:266:in `each'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/beefcake-1.1.0/lib/beefcake.rb:266:in `assign'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/beefcake-1.1.0/lib/beefcake.rb:246:in `initialize'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:201:in `new'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:201:in `serialize_inner_map'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:107:in `inner_serialize'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:85:in `block in inner_serialize_group'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:84:in `map'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:84:in `inner_serialize_group'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:189:in `serialize_map'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:74:in `serialize_group'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:58:in `serialize_wrap'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:52:in `serialize'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client/beefcake/crdt_operator.rb:28:in `operate'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/map.rb:106:in `block in write_operations'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/base.rb:115:in `block in operator'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client.rb:357:in `block in recover_from'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/innertube-1.0.2/lib/innertube.rb:127:in `take'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client.rb:355:in `recover_from'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/client.rb:327:in `protobuffs'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/base.rb:104:in `backend'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/base.rb:114:in `operator'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/map.rb:105:in `write_operations'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/map.rb:56:in `batch'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/map.rb:64:in `operate'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/typed_collection.rb:134:in `operate'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/inner_map.rb:29:in `operate'
        from /Users/kazuhiro/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/riak-client-2.2.1/lib/riak/crdt/typed_collection.rb:123:in `delete'
        from ./map-test.rb:12:in `<main>'
```

### CODE REVIEW

File: spec/integration/riak/crdt_spec.rb [hunk 1]
```diff
@@ -219,6 +219,16 @@
...
       end
+
+      it 'deletes nested inner-map' do
+        bag_map = subject.maps['bag']
+        inner_map = bag_map.maps['123']
+        inner_map.registers['name'] = 'f1'
+
+        expect(subject.maps['bag'].maps['123'].registers['name']).to eq 'f1'
+
+        expect{ bag_map.maps.delete('123') }.to_not raise_error
```
Comment (bkerley) [inline spec/integration/riak/crdt_spec.rb]: Can you add an expectation for `bag_map.maps['123'].registers['name']` `to_not be` or something to that effect?

**Comment by ksauzz** [comment]
Thanks for help. I added another test Bryce mentioned, and ran integration tests successfully. :)

```
% bundle exec rake spec:integration
/Users/kazuhiro/.rbenv/versions/2.2.3/bin/ruby -I/Users/kazuhiro/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.3.2/lib:/Users/kazuhiro/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-support-3.3.0/lib /Users/kazuhiro/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.3.2/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb --profile --tag '~slow' --tag integration
Run options:
  include {:focus=>true, :integration=>true}
  exclude {:yes_security=>true, :slow=>true}

Randomized with seed 57996
....................................................................................................................................................................................................................................

Top 10 slowest examples (58.58 seconds, 26.3% of total time):
  Protocol Buffers BeefcakeProtobuffsBackend it should behave like Unified backend API searching fulltext indexes finds indexed documents, returning documents
    10.63 seconds ./spec/support/unified_backend_examples.rb:366
  Protocol Buffers BeefcakeProtobuffsBackend searching yokozuna returns documents with UTF-8 fields (GH #75)
    9.45 seconds ./spec/integration/riak/protobuffs_backends_spec.rb:22
  Encoding and Riak Search with utf-8 strings queries non-weird indexes
    8.9 seconds ./spec/integration/riak/encodings/yz_spec.rb:44
  Yokozuna without any indexes allows index creation
    8.69 seconds ./spec/integration/yokozuna/index_spec.rb:13
  Protocol Buffers BeefcakeProtobuffsBackend it should behave like Unified backend API searching fulltext indexes finds indexed documents, returning ids
    8.61 seconds ./spec/support/unified_backend_examples.rb:353
  Yokozuna with an index has an index list
    5.02 seconds ./spec/integration/yokozuna/index_spec.rb:29
  Object-oriented Search API indexes creates indexes
    4.05 seconds ./spec/integration/riak/search_spec.rb:70
  Yokozuna queries with a schema and indexes supports utf8
    1.12 seconds ./spec/integration/yokozuna/queries_spec.rb:76
  Yokozuna with an index associated with a bucket indexes on object writes
    1.11 seconds ./spec/integration/yokozuna/index_spec.rb:48
  Protocol Buffers interrupted requests fails out when a request is interrupted, and never returns the wrong payload
    1 seconds ./spec/integration/riak/protobuffs/interrupted_request_spec.rb:21

Top 10 slowest example groups:
  Object-oriented Search API
    7.92 seconds average (63.38 seconds / 8 examples) ./spec/integration/riak/search_spec.rb:4
  CRDT Search API
    5.14 seconds average (97.6 seconds / 19 examples) ./spec/integration/riak/crdt_search_spec.rb:4
  Yokozuna
    3.03 seconds average (15.13 seconds / 5 examples) ./spec/integration/yokozuna/index_spec.rb:4
  Encoding and Riak Search
    1.82 seconds average (9.11 seconds / 5 examples) ./spec/integration/riak/encodings/yz_spec.rb:5
  Protocol Buffers
    1.01 seconds average (1.01 seconds / 1 example) ./spec/integration/riak/protobuffs/interrupted_request_spec.rb:4
  Yokozuna queries
    0.51396 seconds average (5.65 seconds / 11 examples) ./spec/integration/yokozuna/queries_spec.rb:5
  Protocol Buffers
    0.34537 seconds average (30.05 seconds / 87 examples) ./spec/integration/riak/protobuffs_backends_spec.rb:3
  Secondary indexes
    0.07596 seconds average (0.45579 seconds / 6 examples) ./spec/integration/riak/secondary_index_spec.rb:3
  Riak::Counter
    0.05784 seconds average (0.05784 seconds / 1 example) ./spec/integration/riak/counters_spec.rb:4
  Conflict resolution
    0.04257 seconds average (0.1277 seconds / 3 examples) ./spec/integration/riak/conflict_resolution_spec.rb:4

Finished in 3 minutes 43.1 seconds (files took 0.33697 seconds to load)
228 examples, 0 failures

Randomized with seed 57996
```

**Comment by lukebakken** [comment]
+1 b0454f48ad35d8c7f1bad990e5ac7a26a3022eb4

**Comment by ksauzz** [comment]
@borshop merge

**Pull request merged**